src/wallet/wallet.cpp

```diff
@@ -4113,20 +4113,25 @@ bool CWallet::UpgradeWallet(int version, bilingual_str& error)
     }
     if (version < prev_version)
     {
-        error = _("Cannot downgrade wallet");
+        error = strprintf(_("Cannot downgrade wallet from version %i to version %i. Wallet version unchanged."), prev_version, version);
+        version = prev_version;
         return false;
     }
 
     LOCK(cs_wallet);
 
     // Do not upgrade versions to any version between HD_SPLIT and FEATURE_PRE_SPLIT_KEYPOOL unless already supporting HD_SPLIT
     if (!CanSupportFeature(FEATURE_HD_SPLIT) && version >= FEATURE_HD_SPLIT && version < FEATURE_PRE_SPLIT_KEYPOOL) {
-        error = _("Cannot upgrade a non HD split wallet without upgrading to support pre split keypool. Please use version 169900 or no version specified.");
+        error = strprintf(_("Cannot upgrade a non HD split wallet from version %i to version %i without upgrading to support pre-split keypool. Please use version %i or no version specified."),
```

Comment:
the translation shouldn't be changed here, if you want this to be backported

Reply:
Oh right (thanks!) and same for the translation just above as well, I suppose
```diff
-        error = _("Cannot downgrade wallet");
+        error = strprintf(_("Cannot downgrade wallet from version %i to version %i. Wallet version unchanged."), prev_version, version);
```

Reply:
If you like the translation changes, you can put them into a separate commit and write "not for backport" in the commit body

Reply:
Done, thanks

Reply:
Thought: do translated strings need to be static? I should look at the implementation.

Reply:
They can only be literals, but may include format specifiers